src/provider/provider_tracking.c

```diff
@@ -161,11 +162,35 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
         return ret;
     }
 
-    umf_result_t ret2 = umfMemoryTrackerAdd(p->hTracker, p->pool, *ptr, size);
-    if (ret2 != UMF_RESULT_SUCCESS) {
-        LOG_ERR("failed to add allocated region to the tracker, ptr = %p, size "
+    // check if the allocation was already added to the tracker
+    // (in case of using ProxyLib)
+    tracker_value_t *value =
+        (tracker_value_t *)critnib_get(p->hTracker->map, *(uintptr_t *)ptr);
+    if (value) {
+        assert(value->pool != p->pool);
+
+        LOG_DEBUG("ptr already exists in the tracker (added by Proxy Lib) - "
+                  "updating value, ptr=%p, size=%zu, old pool: %p, new pool %p",
+                  *ptr, size, (void *)value->pool, (void *)p->pool);
+
+        // the allocation was made by the ProxyLib so we only update the tracker
+        value->pool = p->pool;
```

Comment:
So we assume that ProxyLib will not look for this pointer any more? Is it based on the fact that ProxyLib does not call `umfPoolByPtr()` in its source code?

Reply:
Yes. Also the proxy lib would not free this allocation - this would be done by the user created Pool